Clarify that tools should look for annotations only in recognized locations. (#662)
jspecify/jspecify: docs/docs/spec.md

> defining that there exists a series of enclosing declarations for any type
> usage, not just for a declaration.

> At each declaration that is a [recognized](#recognized-declaration) location,

Reviewer: ```suggestion
At each declaration that is in a [recognized](#recognized-declaration) location,
```

Author: Hmm, I think of the declaration itself as _being_ the recognized location: A method is a recognized location for a `@Target(METHOD)` annotation, etc.

I'll merge this PR as it stands now, but we can continue discussing if you'd like.

## Augmented type of a type usage appearing in code {#augmented-type-of-usage}

> For most type usages in source code or bytecode on which JSpecify nullness

Reviewer: ```suggestion
For most type usages in source code or bytecode where JSpecify nullness
```

annotations are [recognized](#recognized-type-use), this section defines how to

> determine their [augmented types]. Note, however, that rules for specific cases

Reviewer: It won't let me use the "suggestion" button here, but how about:

> This section defines how to determine the [augmented types] of most type usages in source code or bytecode where JSpecify nullness annotations are [recognized].
